contentcuration/contentcuration/static/js/edit_channel/new_channel/views.js

```diff
@@ -79,6 +83,19 @@ var ChannelList  = BaseViews.BaseEditableListView.extend({
 			}
 
 		}
+	},
+	update_channel_collection:function(channel, deleted){
+		if(deleted){
+			window.channels = _.reject(window.channels, function(c){return c.id == channel.id});
+		}else{
+			var match = _.findWhere(window.channels, {id:channel.id});
+			if(match){
+				var index = window.channels.indexOf(match);
+				window.channels[index] = channel.toJSON();
+			}else{
+				window.channels.push(channel.toJSON());
+			}
```

Comment:
this code searches the channel list twice - once using `_.findWhere` and once using `indexOf`.

You could possibly use [`findIndex`](http://underscorejs.org/#findIndex) to return the index on the first pass?